common/src/main/java/com/thoughtworks/go/buildsession/ExecCommandExecutor.java

```diff
@@ -83,6 +84,7 @@ private CommandLine createCommandLine(String cmd) {
         } else {
             commandLine = CommandLine.createCommandLine(cmd);
         }
+        commandLine.withEncoding(new SystemEnvironment().consoleLogCharset());
```

Comment:
@jyotisingh — we'd likely need to inject the charset to all `BuildCommandExecutor` impls, or add it to `BuildSession` to make it available to all executors.

Thoughts?

Reply:
I prefer adding it to the BuildSession. One way would be to pass along the charset with [BuildSettings](https://github.com/gocd/gocd/blob/6889503814329ba12794b9a8a86971afa84bfc3c/common/src/main/java/com/thoughtworks/go/domain/BuildSettings.java#L20) and then use it to construct the `BuildSession`. That way in future too if any other CommandExecutor needs access to commandline, it could just get hold of the charset from `BuildSession`